docs/cmm.md

```diff
@@ -0,0 +1,38 @@
+# Central Machine Management
+
+## Deploy Custom Resource Definitions
+
+CAPI custom resource definitions are not deployed by default and must be applied manually to the Hive cluster.
+
+```
+oc apply -f ./hack/cluster-api/
+```
+
+## Alpha Feature Gate
+
+The `AlphaMachineManagement` feature gate must be enabled in `hiveconfig` before the API will allow `cd.spec.machineManagement.central = {}` to be set for a `ClusterDeployment`.
+
+```
+  spec:
+    featureGates:
+      custom:
+        enabled:
+        - AlphaMachineManagement
+      featureSet: Custom
+```
+
+```
+oc patch hiveconfig hive --patch '{"spec":{"featureGates":{"custom":{"enabled":["AlphaMachineManagement"]},"featureSet":"Custom"}}}' --type=merge
+```
+
+## Request Central Machine Management for ClusterDeployment
+
+CMM can be enabled for a ClusterDeployment by setting `cd.spec.machineManagement.central = {}`
+
+```
+    spec:
+      machineManagement:
+        central: {}
+```
+
+Once CMM is enabled, a `targetNamespace` is generated per `ClusterDeployment` and set as `cd.spec.machineManagement.targetNamespace`. Secrets necessary for creating CAPI `MachineSets` are copied into this `targetNamespace`. Cluster API `MachineSets` and `MachineTemplates` are created in the `targetNamespace`.
```

Comment:
can we include some docs on how to generate these CRDs and keep them in sync to the thirdparty ?

Reply:
I remember `central` being made a dict for future proofing but I'm not sure it is necessary.